packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -308,6 +310,7 @@ class TransactionService {
               }
             }
           }
+          this.setSession(tr)
```

Comment:
The place where we are calling this is wrong. Session should start right when the navigation started not when our transaction is ended and we have to associate them for each transaction that ever happens within that particular session.

Reply:
I actually started with creating the session on the transaction start, but considering that there are some cases in which we might not send (or even end) a transaction, I think this creates inconsistencies, e.g. we start a session for a transaction that we never send to the server and the session would have wrong data.